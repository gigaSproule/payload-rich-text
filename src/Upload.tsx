import type { Options } from "./types";

export interface UploadData {
  type: "upload";
  format: string;
  version: number;
  id: string;
  fields: null;
  relationTo: string;
  value: {
    id: string;
    filename: string;
    mimeType: string;
    filesize: number;
    url: string;
    [key: string]: unknown;
  };
}

export type Props = { data: UploadData; options?: Options };

export const Upload = ({ data, options }: Props) => {
  return options?.upload ? (
    options.upload(data)
  ) : (
    <a href={data.value.url}>{data.value.filename}</a>
  );
};
