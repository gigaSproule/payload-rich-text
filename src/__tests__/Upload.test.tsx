import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../Upload";
import { Upload } from "../Upload";
import { createUploadData } from "./helpers/createUploadData";

describe("Upload", () => {
  it("should render default a if no options provided", () => {
    const uploadData: Props["data"] = createUploadData();
    const { container } = render(<Upload data={uploadData} />);
    expect(container).toMatchSnapshot();
  });

  it("should render default a if options provided without upload", () => {
    const uploadData: Props["data"] = createUploadData();
    const { container } = render(<Upload data={uploadData} options={{}} />);
    expect(container).toMatchSnapshot();
  });

  it("should render custom render if options provided with upload", () => {
    const uploadData: Props["data"] = createUploadData();
    const { container } = render(
      <Upload
        data={uploadData}
        options={{
          upload: (upload) => {
            return <div>Custom Upload - {JSON.stringify(upload)}</div>;
          },
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
