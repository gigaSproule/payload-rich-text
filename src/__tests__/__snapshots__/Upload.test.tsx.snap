// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Upload should render custom render if options provided with upload 1`] = `
<div>
  <div>
    Custom Upload - 
    {"type":"upload","format":"","version":1,"id":"id","fields":null,"relationTo":"image","value":{"id":"image-id","filename":"image.jpg","mimeType":"image/jpeg","filesize":1,"url":"https://www.example.com/image.jpg","some":"custom","block":"fields"}}
  </div>
</div>
`;

exports[`Upload should render default a if no options provided 1`] = `
<div>
  <a
    href="https://www.example.com/image.jpg"
  >
    image.jpg
  </a>
</div>
`;

exports[`Upload should render default a if options provided without upload 1`] = `
<div>
  <a
    href="https://www.example.com/image.jpg"
  >
    image.jpg
  </a>
</div>
`;
